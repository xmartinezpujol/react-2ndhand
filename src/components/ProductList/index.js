import React from 'react';

import Logo from '../../components/Shared/Logo';
import ProductCard from '../../components/ProductCard';

import {
  ListContainer,
  ListPageWrapper,
  LogoWrapper,
} from './styles';

const ProductList = ({
  products,
}: {
  products: ProductType[]
}) => (
    <ListPageWrapper>
      <ListContainer>
        <LogoWrapper>
          <Logo />
        </LogoWrapper>
        {products && products.map(ad => (
          <ProductCard key={ad.id} product={ad} />
        ))}
      </ListContainer>
    </ListPageWrapper>
);

export default ProductList;
