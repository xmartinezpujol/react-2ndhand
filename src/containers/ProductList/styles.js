import styled from 'styled-components';

export const ListPageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
`;

export const ListContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: center;
  width: 100%;
  padding: 20px;
  @media (min-width: 768px) {
    flex-direction: row;
    padding: 30px;
    justify-content: space-between;
  }
  
  @media (min-width: 992px) {
    max-width: 1200px;
  }
`;

export const LogoWrapper = styled.div`
  display: flex;
  width: 100%;
  margin-bottom: 20px;
`;
