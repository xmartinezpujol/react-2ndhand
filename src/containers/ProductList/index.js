import React from 'react';

import Logo from '../../components/Shared/Logo';
import ProductCard from '../../components/ProductCard';

import { ListContainer, ListPageWrapper, LogoWrapper } from './styles';

type Props = {
  products: ProductType[]
};
type State = {};

class ProductList extends React.Component<Props, State> {
  render() {
    const { products } = this.props;
    return (
      <ListPageWrapper>
        <ListContainer>
          <LogoWrapper>
            <Logo />
          </LogoWrapper>
          {products && products.map(ad => (
            <ProductCard key={ad.id} data={ad} />
          ))}
        </ListContainer>
      </ListPageWrapper>
    );
  }
}

export default ProductList;
