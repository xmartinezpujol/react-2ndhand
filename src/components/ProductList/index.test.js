import React from 'react';
import { mount } from 'enzyme';
import { StaticRouter } from 'react-router';
import renderer from 'react-test-renderer';

// $FlowFixMe
import Ads from '../../mocks/ads.json';

import ProductList from './index';

describe('ProductList', () => {
  const addsByPage = Ads.map(page => page.ads);
  const allAds = [].concat.apply([], addsByPage);
  it('renders correctly', () => {
    const context = {};
    const tree = renderer.create(
      <StaticRouter location="/list" context={context}>
        <ProductList products={allAds} />
      </StaticRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders products when data is received', () => {
    const context = {};
    const twoProducts = allAds.slice(0, 2);
    const wrapper = mount(
      <StaticRouter location="/list" context={context}>
        <ProductList products={twoProducts} />
      </StaticRouter>
    );
    expect(wrapper.find('ProductCard').length).toBe(2);
  });
  it('renders no products when data is empty', () => {
    const context = {};
    const products = [];
    const wrapper = mount(
      <StaticRouter location="/list" context={context}>
        <ProductList products={products} />
      </StaticRouter>
    );
    expect(wrapper.find('ProductCard').length).toBe(0);
  });
});
