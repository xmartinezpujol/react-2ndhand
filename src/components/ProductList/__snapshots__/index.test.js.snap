// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders correctly 1`] = `
<div
  className="styles__ListPageWrapper-yxzh6s-0 bmMZQI"
>
  <div
    className="styles__ListContainer-yxzh6s-1 eJyhsY"
  >
    <div
      className="styles__LogoWrapper-yxzh6s-2 ZRYKe"
    >
      <div
        className="logo"
      >
        <svg
          height="84"
          viewBox="0 0 84 84"
          width="84"
        >
          <g>
            <path
              className="color-1"
              d="M40.86,0a42,42,0,0,1,38.28,61.6A42,42,0,0,1,22.45,79.16a42.53,42.53,0,0,1-12.27-9.77A42,42,0,0,1,1.09,51.6a42.42,42.42,0,0,1,.53-21.21A41.73,41.73,0,0,1,8.15,17.15,42.46,42.46,0,0,1,21.23,5.49,41.47,41.47,0,0,1,40.86,0ZM40,1.34A40.49,40.49,0,0,0,18.09,9,41.24,41.24,0,0,0,5.83,23.25,40.64,40.64,0,0,0,9.34,66.31a41.45,41.45,0,0,0,10.9,10.1A40.73,40.73,0,0,0,77.57,61.83a40.6,40.6,0,0,0-4.48-46.06A41,41,0,0,0,61.38,6.22,39.69,39.69,0,0,0,40,1.34Z"
            />
          </g>
          <g>
            <path
              className="color-2"
              d="M34.41,22.32a30.21,30.21,0,0,1,3.39-.05,7.28,7.28,0,0,1,2.1.13A1.75,1.75,0,0,1,41,24.11v16.1c0,.51-.08,1.09.28,1.49s.63.25,1,.28h6.2a1.85,1.85,0,0,1,2,1.92c0,1.19,0,2.38,0,3.57a1.78,1.78,0,0,1-1.87,1.67H34.74A1.84,1.84,0,0,1,33,47.22V24.11a1.94,1.94,0,0,1,1.44-1.8Z"
            />
          </g>
          <g>
            <path
              className="color-3"
              d="M70,27a4.25,4.25,0,0,1,2.91-1.49,4.19,4.19,0,0,1,1.47.28,3.72,3.72,0,0,1,.3,2.76,9.67,9.67,0,0,1-2.28,3.9c-1.67,2.1-3.19,4.3-4.73,6.5-.33.48-.68.94-1,1.44-.18.18,0,.35.13.53,3.16,4.63,6.3,9.29,9.46,13.92a4.63,4.63,0,0,1,1.16,2.83,1.78,1.78,0,0,1-1.21,1.47,25.24,25.24,0,0,1-6,1.67A3.28,3.28,0,0,1,66.82,59c-1.72-3.62-3.59-7.14-5.41-10.71-2.18,3.21-4.3,6.48-6.43,9.74a4.1,4.1,0,0,1-1.59,1.42,1.35,1.35,0,0,1-1.32-.08,5.17,5.17,0,0,1-1-1.09,31.55,31.55,0,0,1-2-3.14A4,4,0,0,1,48.86,52a7.24,7.24,0,0,1,1.47-2.48q3.64-4.18,7.31-8.33c-1.67-3.26-3.39-6.48-5.06-9.72A5.53,5.53,0,0,1,52,28.54a2.37,2.37,0,0,1,1.14-2,3.61,3.61,0,0,1,3.44-.1,3.49,3.49,0,0,1,1.26,1.24l5.08,7.49C65.28,32.44,67.63,29.71,70,27Z"
            />
          </g>
          <g>
            <path
              className="color-4"
              d="M16.37,26.62A12.1,12.1,0,0,1,24,28.21a15.27,15.27,0,0,1,6.3,7.67,18.66,18.66,0,0,1,.53,11.74,15.82,15.82,0,0,1-6.1,8.81,12.14,12.14,0,0,1-6.81,2.18,12.29,12.29,0,0,1-7.18-2.38,16.15,16.15,0,0,1-6-9.57A18.38,18.38,0,0,1,6.43,33.78a14.54,14.54,0,0,1,6.07-6,12.27,12.27,0,0,1,3.87-1.19Zm.4,5.14a6.07,6.07,0,0,0-3.19,2.58,13.73,13.73,0,0,0-2,5.75,17.43,17.43,0,0,0,1,8.88,8.26,8.26,0,0,0,3.06,3.92,4,4,0,0,0,3.74.33,6.29,6.29,0,0,0,2.71-2.43,14.25,14.25,0,0,0,2-5.69,17.41,17.41,0,0,0-1.06-9,8.94,8.94,0,0,0-2.86-3.77,4.13,4.13,0,0,0-3.44-.56Z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      className="ui link card styles__CardWrapper-sc-13wxosj-0 ksZAYB"
      onClick={[Function]}
    >
      <a
        className="content"
        href="/list/205799123"
        onClick={[Function]}
      >
        <div
          className="header"
        >
          <div
            className="styles__Title-sc-13wxosj-4 gXnukw"
          >
            iphone 5s 16gb silver bialy stan dbd Pl dystrybucja 100% sprawny
          </div>
        </div>
        <div
          className="meta"
        >
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Marka
            <div
              className="detail"
            >
              iPhone
            </div>
          </div>
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Stan
            <div
              className="detail"
            >
              Używane
            </div>
          </div>
        </div>
        <div
          className="description styles__CardDescription-sc-13wxosj-1 jJIAfJ"
        >
          Sprzedam Iphone 5s w bdb stanie technicznym i wizualnym (jak na foto) telefon jest w 100 % sprawny wszystko dziala nie zawiesza sie i nie gubi sieci. Wizualnie jak na zdjeciach posiada normalne slady urzytkowania,
Pl dystrybucja telefon fabrycznie bez blokady simlock. 
Telefon sprzedaje w kmpl 
W sk...
        </div>
      </a>
      <div
        className="extra content"
      >
        <i
          aria-hidden="true"
          className="teal time icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          10  lut
        </span>
        <i
          aria-hidden="true"
          className="teal map marker icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          Kobyłka
        </span>
      </div>
    </div>
    <div
      className="ui link card styles__CardWrapper-sc-13wxosj-0 ksZAYB"
      onClick={[Function]}
    >
      <a
        className="content"
        href="/list/136306295"
        onClick={[Function]}
      >
        <div
          className="header"
        >
          <div
            className="styles__Title-sc-13wxosj-4 gXnukw"
          >
            Apple iPhone 4S jak nowy gwarancja, pudelko Warszawa
          </div>
        </div>
        <div
          className="meta"
        >
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Marka
            <div
              className="detail"
            >
              iPhone
            </div>
          </div>
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Stan
            <div
              className="detail"
            >
              Używane
            </div>
          </div>
        </div>
        <div
          className="description styles__CardDescription-sc-13wxosj-1 jJIAfJ"
        >
          Mam do zaoferowania telefon komórkowy Apple iPhone 4S w wersji 8Gb w kolorze białym.Kupiony w polsce rok temu. Stan bardzo dobry, przod od nowosci z folią wiec jest jak nowy. Tyl i boki takze. Technicznie telefon w 100% sprawny. Nigdy nie było z nim żadnych problemów. Jest jak nowy. Cena nie podlega...
        </div>
      </a>
      <div
        className="extra content"
      >
        <i
          aria-hidden="true"
          className="teal time icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          10  lut
        </span>
        <i
          aria-hidden="true"
          className="teal map marker icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          Warszawa Ursynów
        </span>
      </div>
    </div>
    <div
      className="ui link card styles__CardWrapper-sc-13wxosj-0 ksZAYB"
      onClick={[Function]}
    >
      <a
        className="content"
        href="/list/203433957"
        onClick={[Function]}
      >
        <div
          className="header"
        >
          <div
            className="styles__Title-sc-13wxosj-4 gXnukw"
          >
            sprzedam iphona 5s 16 GB space grat
          </div>
        </div>
        <div
          className="meta"
        >
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Marka
            <div
              className="detail"
            >
              iPhone
            </div>
          </div>
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Stan
            <div
              className="detail"
            >
              Używane
            </div>
          </div>
        </div>
        <div
          className="description styles__CardDescription-sc-13wxosj-1 jJIAfJ"
        >
          Sprzedam iphona 5s Telefon bez simlocka działa w każdej sieci. Normalne śladu użytkowania. Z lewej strony ekranu na białym tle widać delikatną żółtą plamkę
        </div>
      </a>
      <div
        className="extra content"
      >
        <i
          aria-hidden="true"
          className="teal time icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          1  lut
        </span>
        <i
          aria-hidden="true"
          className="teal map marker icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          Wrocław Śródmieście
        </span>
      </div>
    </div>
    <div
      className="ui link card styles__CardWrapper-sc-13wxosj-0 ksZAYB"
      onClick={[Function]}
    >
      <a
        className="content"
        href="/list/203387821"
        onClick={[Function]}
      >
        <div
          className="header"
        >
          <div
            className="styles__Title-sc-13wxosj-4 gXnukw"
          >
            Idealny IPHONE 5S GOLD + Gratisy
          </div>
        </div>
        <div
          className="meta"
        >
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Marka
            <div
              className="detail"
            >
              iPhone
            </div>
          </div>
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Stan
            <div
              className="detail"
            >
              Używane
            </div>
          </div>
        </div>
        <div
          className="description styles__CardDescription-sc-13wxosj-1 jJIAfJ"
        >
          Witam dziś mam na sprzedaż idealnego iphona 5s w kolorze złotym .  Jego stan oceniam na idealny. Wygląda jak nowy. Kolor złoty. Wersja 16 GB.  Na gwarancji Jeszcze z 3- miesiące.  Od początku szkło. Ostatnio nałożone nowe szkło z 3MK( flexible Glass 3D które chroni cały telefon z każdej strony)  dod...
        </div>
      </a>
      <div
        className="extra content"
      >
        <i
          aria-hidden="true"
          className="teal time icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          dzisiaj 14:51
        </span>
        <i
          aria-hidden="true"
          className="teal map marker icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          Katowice Śródmieście
        </span>
      </div>
    </div>
    <div
      className="ui link card styles__CardWrapper-sc-13wxosj-0 ksZAYB"
      onClick={[Function]}
    >
      <a
        className="content"
        href="/list/203512163"
        onClick={[Function]}
      >
        <div
          className="header"
        >
          <div
            className="styles__Title-sc-13wxosj-4 gXnukw"
          >
            Iphone 4s biały czarny 8 gb, 16gb, 32gb
          </div>
        </div>
        <div
          className="meta"
        >
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Marka
            <div
              className="detail"
            >
              iPhone
            </div>
          </div>
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Stan
            <div
              className="detail"
            >
              Używane
            </div>
          </div>
        </div>
        <div
          className="description styles__CardDescription-sc-13wxosj-1 jJIAfJ"
        >
          Mam do zaoferowania Iphone 4s

Telefon oryginalny z europejskiej dystrybucji, nie żaden chińczyk jakich pełno na rynku.

Telefon bez simlocka i icloud.

W komplecie ładowarka, usb .

Do wyboru kolor biały czarny , 8gb 16gb, 32gb

Gwarancja producenta  serwisowa 6msc.
Paragon lub FV Vat-ma...
        </div>
      </a>
      <div
        className="extra content"
      >
        <i
          aria-hidden="true"
          className="teal time icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          dzisiaj 14:51
        </span>
        <i
          aria-hidden="true"
          className="teal map marker icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          Kielce
        </span>
      </div>
    </div>
    <div
      className="ui link card styles__CardWrapper-sc-13wxosj-0 ksZAYB"
      onClick={[Function]}
    >
      <a
        className="content"
        href="/list/206966241"
        onClick={[Function]}
      >
        <div
          className="header"
        >
          <div
            className="styles__Title-sc-13wxosj-4 gXnukw"
          >
            Zamienię Sony Xperia M4 Aqua
          </div>
        </div>
        <div
          className="meta"
        >
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Marka
            <div
              className="detail"
            >
              iPhone
            </div>
          </div>
          <div
            className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
            onClick={[Function]}
          >
            Stan
            <div
              className="detail"
            >
              Używane
            </div>
          </div>
        </div>
        <div
          className="description styles__CardDescription-sc-13wxosj-1 jJIAfJ"
        >
          Zamienię Sony Xperia m4 Aqua, Telefon półroczny w stanie idealnym. Od nowości naklejona szyba Pancerna. Telefon noszony w eui Nilkin. Do telefonu oryginalne pudełku, nieużywane słuchawki oraz półtora roku gwarancji.
Zamienię na IPhona 5s, 6, 6s z odpowiednią dopłatą z mojej strony. Najlepiej z gwar...
        </div>
      </a>
      <div
        className="extra content"
      >
        <i
          aria-hidden="true"
          className="teal time icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          dzisiaj 14:50
        </span>
        <i
          aria-hidden="true"
          className="teal map marker icon"
          onClick={[Function]}
        />
        <span
          className="styles__CardInfo-sc-13wxosj-2 hDerPd"
        >
          Częstochowa Błeszno
        </span>
      </div>
    </div>
  </div>
</div>
`;
